File: baicaio_code/baicaio/baicaio/Extension/Image+Color/UIImage+Color.m
//
//  UIImage+Color.m
//  B2C_iPhone_V4
//
//  Created by JimBo on 15/6/30.
//  Copyright (c) 2015年 edu24ol. All rights reserved.
//

#import "UIImage+Color.h"

@implementation UIImage (Color)

+(UIImage *)imageWithColor:(UIColor *)aColor{
    return [UIImage imageWithColor:aColor withFrame:CGRectMake(0, 0, 1, 1)];
}

+(UIImage *)imageWithColor:(UIColor *)aColor withFrame:(CGRect)aFrame{
    UIGraphicsBeginImageContext(aFrame.size);
    CGContextRef context = UIGraphicsGetCurrentContext();
    CGContextSetFillColorWithColor(context, [aColor CGColor]);
    CGContextFillRect(context, aFrame);
    UIImage *img = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    return img;
}


@end

File: baicaio_code/baicaio/baicaio/Extension/UIImage+Color.swift
//
//  UIImage+Color.swift
//  baicaio
//
//  Created by JimBo on 15/12/25.
//  Copyright © 2015年 JimBo. All rights reserved.
//

import Foundation
public extension UIImage{

    public static func imageWithColor(color:UIColor,withFrame frame:CGRect)->UIImage{
        UIGraphicsBeginImageContext(frame.size)
        let context = UIGraphicsGetCurrentContext()
        CGContextSetFillColorWithColor(context, color.CGColor)
        CGContextFillRect(context, frame)
        let img = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        return img
    }
    
    public static func imageWithColor(color:UIColor)->UIImage{
        return UIImage.imageWithColor(color, withFrame: CGRectMake(0, 0, 1, 1))
    }

}